src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -317,18 +312,30 @@ object ArrayBuffer extends StrictOptimizedSeqFactory[ArrayBuffer] {
   }
 }
 
-final class ArrayBufferView[A] private[mutable](val array: Array[AnyRef], val length: Int, mutationCount: () => Int)
+// TODO: use `CheckedIndexedSeqView.Id` once we can change the return type of `ArrayBuffer#view`
+final class ArrayBufferView[A] private[mutable](underlying: ArrayBuffer[A], mutationCount: () => Int)
   extends AbstractIndexedSeqView[A] {
   @deprecated("never intended to be public; call ArrayBuffer#view instead", since = "2.13.6")
   def this(array: Array[AnyRef], length: Int) = {
     // this won't actually track mutation, but it would be a pain to have the implementation
     // check if we have a method to get the current mutation count or not on every method and
     // change what it does based on that. hopefully no one ever calls this.
-    this(array, length, () => 0)
+    this({
+      val _array = array
+      val _length = length
+      new ArrayBuffer[A](0) {
+        this.array = _array
+        this.size0 = _length
+      }
+    }, () => 0)
   }
 
+  @deprecated("never intended to be public", since = "2.13.6")
```

Comment:
this PR has been open too long 😅

```suggestion
  @deprecated("never intended to be public", since = "2.13.7")
```